File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int max=0;
        int i=0; int j=height.length-1;
        int curr=0;
        while(i<j)
        {
            curr=Math.min(height[i], height[j])*(j-i);
            max=Math.max(max,curr);
            if(height[i]<=height[j])
            i++;
            else
            j--;

        }
        return max;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
  let max=0;
  let i=0; let j=height.length-1;
  let curr=0;
  while(i<j)
  {
    curr=Math.min(height[i], height[j])*(j-i);
    max=Math.max(curr,max);
    if(height[i]<=height[j])
     i++;
     else
     j--;
  }
  return max;
};